### PR TITLE
Issue #201 assemble-maven-repository respect skip from maven-deploy-plugin

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### CODE REVIEW

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (laeubi) [inline tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java:439]: same as above

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (laeubi) [inline tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java:472]: same as above...

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (laeubi) [inline tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java:451]: Maybe this is better placed in a common helper-class @mickaelistria any idea what one would be suitable?

File: .gitignore
Comment (laeubi) [inline .gitignore:7]: Side note: Are you using InteliJ? Would be cool to add a little note to the contribution guide how to setup tycho with InteliJ then (as a separate PR)!

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (laeubi) [inline tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java:207]: Now we are adding more (and maybe more in the future) conditions and checks I think it would be good to extract the whole block into a method:

```
if (includeReactor) {
            resolveReactorProjects(mavenProject, bundles, advices);
}
```

**Review by laeubi — CHANGES_REQUESTED** [review]
Thanks for the PR, this looks fine to me but some adjustments would be good.

1. I have already prepared an (incomplete) integration test [here](https://github.com/laeubi/tycho/commit/29ef2f921105710132e0810828ccfcb75a85af9b) , maybe you can integrate this into your PR and complete it with the different cases you like to have covered?
2. I have added some comments regarding code structure and logging

**Comment by olamy** [comment]
> 1. I have already prepared an (incomplete) integration test [here](https://github.com/laeubi/tycho/commit/29ef2f921105710132e0810828ccfcb75a85af9b) , maybe you can integrate this into your PR and complete it with the different cases you like to have covered?
Great I will cherry-pick there.

File: .gitignore
Comment (olamy) [reply .gitignore:7]: yes I'm using InteliJ. Frankly nothing special/magic I just imported the project as a Maven project.
Not sure if that worth a PR. Bear in mind I'm a Maven cli guy so not running test via IDE :)

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (olamy) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java:433]: there is no trace level in the Maven plugin logging API.

Comment (laeubi) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java:433]: okay then either remove (if only used for debugging) or replace with a user understandable string, tycho it is sometimes usefull to run a tycho build with -X to debug complexer dependency scenarios and thus debugmessages should be at least descriptive to the user as well.

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (olamy) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java:474]: @laeubi wdym with above?

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (olamy) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java]: fixed

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (olamy) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java]: fixed

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (laeubi) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java]: this refers to "Maybe this is better placed in a common helper-class" but I don't see this as a blocker just thought it might better be placed in a common place, I'll check if I found a suitable place for this

Comment (laeubi) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java]: org.eclipse.tycho.core.utils.MavenSessionUtils might be a good place so this code could be reused by other mojos as well...

File: tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java
Comment (laeubi) [reply tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java]: org.eclipse.tycho.core.utils.MavenSessionUtils might be a good place so this code could be reused by other mojos as well...

**Comment by laeubi** [comment]
`TargetPlatformLocationsTest.testMavenLocationRepositories` test is flaky atm (no idea why), so don'T mind ignoring it for the moment, this is not related to this change.

**Review by laeubi — APPROVED** [review]
This looks good to me. Is there anything left from your side before we merge this?

**Comment by olamy** [comment]
@laeubi all looks good to me.

**Comment by mickaelistria** [comment]
Please allow me some more time to discuss and review before merging as I'm not confident coupling with other properties owned by other mojos and parsing poms instead of processing model is a good idea in general.
I'm on PTO this week and will look at it next week.

**Comment by laeubi** [comment]
>  parsing poms instead of processing model

can you explain where poms are parsed here?

**Comment by laeubi** [comment]
@mickaelistria finally the build succeeds, so this is "ready to review" if you are. I just wanted to note that as this is specifically enpower maven projects to publish p2-sites in a maven way it makes perfect sense to me to have some specialties.

To clarify this a bit:

1. This **only** applies to **reactor projects** if these are included in the site as part of a build, so site+projects are part of the same reactor build and there are no artifacts
2. It does not apply to any declared **dependencies** that are maybe included (as these have to be deployed anyways before they could be used)
3. Should we anytime later think it would be good to include reactor projects that are never deployed we can easily add a switch for this (or we can add it now if that helps vanish your concerns)
4. This is made for supporting common practice (not specific to jetty project) as the alternative would be to add some kind of sophisticated filtering options that require to duplicate configuration already there from a users point of view

**Comment by olamy** [comment]
side note such skip already exists for javadoc plugin. https://github.com/apache/maven-javadoc-plugin/pull/36 (https://issues.apache.org/jira/browse/MJAVADOC-613)

**Comment by laeubi** [comment]
@mickaelistria do you think we can proceed here? Do you need any further information/background?

**Comment by mickaelistria** [comment]
I'm OK about adding this functionality. However, are all the commits in the PR related? Please make sure only the relevant changes are merged (and squashed).

**Comment by olamy** [comment]
everything is related and squash done.

**Review by laeubi — APPROVED** [review]
Looks good to me. I'll merge this then once the build has passed.

**Comment by mickaelistria** [comment]
Please make sure to add a note in the RELEASE_NOTES to describe what such feature can help with.

**Comment by laeubi** [comment]
It was already added to the [release-notes](https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md#create-p2-repository-referencing-maven-artifacts)  and shows up in the [documentation](https://www.eclipse.org/tycho/sitedocs/tycho-p2/tycho-p2-repository-plugin/assemble-maven-repository-mojo.html)

**Pull request merged**
